src/Operators/finitedifference.jl

```diff
@@ -1229,7 +1232,7 @@ function stencil_right_boundary(
     arg,
 )
     space = axes(arg)
-    @assert idx <= right_face_boundary_idx(space) + 1
+    @assert idx <= right_face_boundary_idx(space) - 1
 
```

Comment:
@simonbyrne I think I spotted a bug. Or at least, here we were not using the same index as we are in the `FirstOrderOneSided` case, on line 1188 above

Reply:
Good catch!